WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -61,73 +52,18 @@ public partial class TransactionPreviewViewModel : RoutableViewModel
 
 			NextCommand = ReactiveCommand.CreateFromTask(async () =>
 			{
-				var dialogResult =
-					await NavigateDialog(new EnterPasswordDialogViewModel(""), NavigationTarget.DialogScreen);
+				var transactionAuthorizationInfo = new TransactionAuthorizationInfo(transaction);
+				var authDialog = AuthorizationHelpers.GetAuthorizationDialog(wallet, transactionAuthorizationInfo);
+				var authDialogResult = await NavigateDialog(authDialog, NavigationTarget.DialogScreen);
 
-				if (dialogResult.Kind == DialogResultKind.Normal)
+				if (authDialogResult.Result)
 				{
-					IsBusy = true;
-
-					var passwordValid = await Task.Run(
-						() => PasswordHelper.TryPassword(
-							wallet.KeyManager,
-							dialogResult.Result,
-							out string? compatibilityPasswordUsed));
-
-					if (passwordValid)
-					{
-						// Dequeue any coin-joining coins.
-						await wallet.ChaumianClient.DequeueAllCoinsFromMixAsync(DequeueReason.TransactionBuilding);
-
-						var signedTransaction = transaction.Transaction;
-
-						// If it's a hardware wallet and still has a private key then it's password.
-						if (wallet.KeyManager.IsHardwareWallet && !transaction.Signed)
-						{
-							try
-							{
-								var client = new HwiClient(wallet.Network);
-
-								using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
-								PSBT? signedPsbt = null;
-								try
-								{
-									signedPsbt = await client.SignTxAsync(
-										wallet.KeyManager.MasterFingerprint!.Value,
-										transaction.Psbt,
-										cts.Token);
-								}
-								catch (HwiException ex) when (ex.ErrorCode is not HwiErrorCode.ActionCanceled)
-								{
-									await PinPadViewModel.UnlockAsync();
-
-									signedPsbt = await client.SignTxAsync(
-										wallet.KeyManager.MasterFingerprint!.Value,
-										transaction.Psbt,
-										cts.Token);
-								}
-
-								signedTransaction = signedPsbt.ExtractSmartTransaction(transaction.Transaction);
-							}
-							catch (Exception _)
-							{
-								// probably throw something here?
-							}
-						}
-
-						await broadcaster.SendTransactionAsync(signedTransaction);
-
-						Navigate().Clear();
-
-						IsBusy = false;
-					}
-					else
-					{
-						IsBusy = false;
-						await ShowErrorAsync("Password was incorrect.", "Please try again.", "");
-					}
+					await broadcaster.SendTransactionAsync(transactionAuthorizationInfo.Transaction);
+					Navigate().Clear();
 				}
 			});
+
+			EnableAutoBusyOn(NextCommand);
```

Comment:
We want the transaction info to be visible behind the dialog to give context.   Busy should only be show when its finally authorised and transmitting, hence the need for manual control.

Reply:
yeah true, done